rmw_connext_shared_cpp/include/rmw_connext_shared_cpp/wait.hpp

```diff
@@ -363,13 +456,23 @@ wait(
       if (!(j < active_conditions->length())) {
         clients->clients[i] = 0;
       }
-      DDS::ReturnCode_t retcode = dds_wait_set->detach_condition(read_condition);
-      if (retcode != DDS::RETCODE_OK) {
-        RMW_SET_ERROR_MSG("Failed to get detach condition from wait set");
-        return RMW_RET_ERROR;
+      rmw_ret_t ret_code = __detach_condition(dds_wait_set, read_condition);
+      if (ret_code != RMW_RET_OK) {
+        return ret_code;
       }
     }
   }
+  {
+    rmw_ret_t ret_code = __handle_active_event_conditions(events);
+    if (ret_code != RMW_RET_OK) {
+      return ret_code;
+    }
+    // reset the status conditions to none.
+    for (auto status_condition : status_conditions) {
+      status_condition->set_enabled_statuses(DDS_STATUS_MASK_NONE);
```

Comment:
Are we disabling all the status checking between calls to wait? If so, these should all be left enabled even while  wait is not executing so we catch events that occur between.

Reply:
@ross-desmond can you please weigh in?